lib/backend/model/resource.go

```diff
@@ -87,6 +87,17 @@ type ResourceListOptions struct {
 	Namespace string
 	// The resource kind.
 	Kind string
+	// Whether the name is prefix rather than the full name.
+	Prefix bool
+}
+
+// If the Kind, Namespace and Name are specified, but the Name is a prefix then the
+// last segment of this path is a prefix.
+func (options ResourceListOptions) ListOptionsLastSegmentIsPrefix() bool {
+	return len(options.Kind) != 0 &&
+		(len(options.Namespace) != 0 || !namespace.IsNamespaced(options.Kind)) &&
```

Comment:
I wonder if we can add some validation (doesn't have to be now), to make sure this is only for WEPs, NPs and GNP kinds? or is that too much? WDYT?

Reply:
For etcdv3 this function will be completely generic, so we don't need to limit for certain resource types.

Reply:
Actually thinking about it, for O.S. in KDD, only WEP prefix is supported.  I'll update the comment in the ListOptions.